Sming/Wiring/Print.cpp

```diff
@@ -197,31 +198,63 @@ size_t Print::println(const Printable &p)
   return n;
 }
 
-size_t Print::printf(const char *fmt, ...)
+size_t Print::vprintf(const char *fmt, va_list va) 
 {
-	size_t sz = 0;
-	size_t buffSize = INITIAL_PRINTF_BUFFSIZE;
-	bool retry = false;
-	do {
-		char tempBuff[buffSize];
-		va_list va;
-		va_start(va, fmt);
-		sz = m_vsnprintf(tempBuff,buffSize, fmt, va);
-		va_end(va);
-		if (sz > (buffSize -1))
-		{
-			buffSize = sz + 1; // Leave room for terminating null char
-			retry = true;
-		}
-		else
-		{
-			if (sz > 0)
-			{
-				write(tempBuff,sz);
-			}
+	size_t size = INITIAL_PRINTF_BUFFSIZE - 1;
+	
+	//	need to retry until size is big enough as m_vsnprintf is currently broken
+	//	(returns maxLen instead of required size)
+	while (1) {
+		char buffer[size + 1];
+		auto sz = m_vsnprintf(buffer, sizeof(buffer), fmt, va);
+		if (sz <= size) {
+			write(buffer, sz);
 			return sz;
 		}
-	} while (retry);
+		size = sz;	
+	}
+}
+
+size_t Print::printf(const char *fmt, ...)
+{
+	va_list va;
+	va_start(va, fmt);
+	auto cnt = vprintf(fmt, va);
```

Comment:
Please only use auto when hiding an ugly/templetized type specifier. In this case it is size_t. IMHO it's faster to read and debug if you know the types.